web/src/settings_banner.ts

```diff
@@ -0,0 +1,164 @@
+import type {Banner} from "./banners.ts";
+import {$t} from "./i18n.ts";
+import * as settings_data from "./settings_data.ts";
+
+const UPGRADE_ACCESS_BANNER: Banner = {
+    intent: "info",
+    label: $t({defaultMessage: "Available on Zulip Cloud Standard."}),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Upgrade to access"}),
+            custom_classes: "request-upgrade",
+            attention: "quiet",
+        },
+    ],
+    close_button: false,
+};
+
+const UPGRADE_BANNER: Banner = {
+    intent: "neutral",
+    label: $t({defaultMessage: "Available on Zulip Cloud Standard."}),
+    buttons: [],
+    close_button: false,
+};
+
+const UPGRADE_OR_SPONSORSHIP_BANNER: Banner = {
+    intent: "info",
+    label: $t({
+        defaultMessage: "Available on Zulip Cloud Standard.",
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Upgrade"}),
+            custom_classes: "request-upgrade",
+            attention: "quiet",
+        },
+        {
+            label: $t({defaultMessage: "Request sponsorship"}),
+            custom_classes: "request-sponsorship",
+            attention: "borderless",
+        },
+    ],
+    close_button: false,
+};
+
+const SPONSORSHIP_BANNER: Banner = {
+    intent: "neutral",
+    label: $t({defaultMessage: "Available on Zulip Cloud Standard."}),
+    buttons: [],
+    close_button: false,
+};
+
+const GROUP_INFO_BANNER: Banner = {
+    intent: "info",
+    label: $t({
+        defaultMessage:
+            "User groups offer a flexible way to manage permissions in your organization.",
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Learn more"}),
+            custom_classes: "user-groups-info",
+            attention: "quiet",
+        },
+    ],
+    close_button: false,
+};
+
+const STREAM_INFO_BANNER: Banner = {
+    intent: "info",
+    label: $t({
+        defaultMessage: "Channels organize conversations based on who needs to see them.",
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Learn more"}),
+            custom_classes: "stream-info",
+            attention: "quiet",
+        },
+    ],
+    close_button: false,
+};
+
+const MOBILE_PUSH_NOTIFICATION_BANNER: Banner = {
+    intent: "warning",
+    label: $t({
+        defaultMessage: "Mobile push notifications are not enabled on this server.",
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Learn more"}),
+            custom_classes: "mobile-push-notification-info",
+            attention: "quiet",
+        },
+    ],
+    close_button: false,
+};
+
+function initialize_upgrade_banners(): void {
+    const has_billing_access = settings_data.user_has_billing_access();
+    const $org_upgrade_container = $(".organization-upgrade-banners-container");
+    if ($org_upgrade_container.length > 0 && has_billing_access) {
+        banners.open(UPGRADE_ACCESS_BANNER, $org_upgrade_container);
+    } else {
```

Comment:
Upgrade banner is not visible for message retention setting in stream settings UI.